drivers/fastboot/fb_fsl/fb_fsl_partitions.c

```diff
@@ -250,15 +265,59 @@ static int _fastboot_parts_load_from_ptable(void)
 	strcpy(ptable[PTN_BOOTLOADER2_INDEX].name, FASTBOOT_PARTITION_BOOTLOADER2);
 	ptable[PTN_BOOTLOADER2_INDEX].start =
 				(CONFIG_FSL_FASTBOOT_BOOTLOADER2_OFFSET * dev_desc->blksz) / dev_desc->blksz;
-	ptable[PTN_BOOTLOADER2_INDEX].length =
-				(ANDROID_BOOTLOADER_SIZE - (CONFIG_FSL_FASTBOOT_BOOTLOADER2_OFFSET * dev_desc->blksz))
-				/ dev_desc->blksz;
+	ptable[PTN_BOOTLOADER2_INDEX].length = (ANDROID_BOOTLOADER_SIZE -
+		(CONFIG_FSL_FASTBOOT_BOOTLOADER2_OFFSET * dev_desc->blksz))/ dev_desc->blksz;
 	ptable[PTN_BOOTLOADER2_INDEX].partition_id = boot_partition;
 	ptable[PTN_BOOTLOADER2_INDEX].flags = FASTBOOT_PTENTRY_FLAGS_UNERASEABLE;
 	strcpy(ptable[PTN_BOOTLOADER2_INDEX].fstype, "raw");
 	last_bootloader_partition = PTN_BOOTLOADER2_INDEX;
 #endif
 
+#ifdef CONFIG_FSL_FASTBOOT_BOOTLOADER_SECONDARY
+	/*
+	 * Bootloader images for Secondary rendundant boot
+	 */
+	strcpy(ptable[PTN_BOOTLOADER_SECONDARY_INDEX].name, FASTBOOT_PARTITION_BOOTLOADER_SECONDARY);
+	ptable[PTN_BOOTLOADER_SECONDARY_INDEX].start = CONFIG_SECONDARY_BOOT_SECTOR_OFFSET +
```

Comment:
In theory this is the same as `ptable[PTN_BOOTLOADER_INDEX].start + CONFIG_SECONDARY_BOOT_SECTOR_OFFSET`